### PR TITLE
Improvements to run and stop to make the output more usable

### PR DESCRIPTION
**Pull request by cliftonc**
Related to #174

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -374,7 +374,7 @@ Bosco.prototype.getTeam = function() {
   var teamConfig = self.config.get('teams');
   var currentTeam;
   _.keys(teamConfig).forEach(function(team) {
-    if (teamConfig[team].path === self.options.workspace) {
+    if (self.options.workspace.indexOf(teamConfig[team].path) >= 0) {
```
Comment (cliftonc) [inline index.js]: This is sort of unrelated but means team detection works as expected?  I did think we fixed this however previously ...

File: src/RunListHelper.js [hunk 2]
```diff
@@ -103,20 +103,20 @@ function getRepoRunList(/* Same arguments as above */) {
 
 function getServiceConfigFromGithub(bosco, repo, next) {
   var team = bosco.getTeam();
+  if (team === 'no-team') {
+    return next();
```
Comment (cliftonc) [inline src/RunListHelper.js]: This avoids spamming people when no-team and trying to get from github.

**Comment by cliftonc** [comment]
@bettiolo can you review?

File: commands/run.js [hunk 3]
```diff
@@ -128,7 +131,12 @@ function cmd(bosco, args, allDone) {
         } else {
           runService(runConfig, asyncMapCb);
         }
-      }, cb);
+      }, function(err) {
+        if (missingDependencies.length > 0) {
+          bosco.warn('Unable to start dependencies: ' + missingDependencies.join(',').cyan);
```
Comment (bettiolo) [inline commands/run.js]: Should we make this an error instead? It is not very visible in the output.

File: commands/run.js [hunk 4]
```diff
@@ -137,9 +145,9 @@ function cmd(bosco, args, allDone) {
       var nodeServices = _.filter(runList, function(i) { return i.service.type === 'node' && _.startsWith('service-', i.name); });
       var nodeApps = _.filter(runList, function(i) { return i.service.type === 'node' && !_.startsWith('service-', i.name); });
       async.mapSeries([
-          {services: infraServices, limit: bosco.concurrency.cpu},
-          {services: nodeServices, limit: bosco.concurrency.cpu},
-          {services: nodeApps, limit: bosco.concurrency.cpu},
+          {services: infraServices, type: 'general', limit: bosco.concurrency.cpu},
```
Comment (bettiolo) [inline commands/run.js]: `general` was quite a confusing term:

```
Bosco: Launching 11 general processes with parallel limit of 7 ...
```

File: index.js (hunk 1 above)
Comment (bettiolo) [inline index.js]: In fact I was having problems with that on a different project.

File: commands/run.js (hunk 4 above)
Comment (cliftonc) [inline commands/run.js]: Any suggestions otherwise?

File: commands/run.js (hunk 3 above)
Comment (cliftonc) [inline commands/run.js]: Yep - fine to make an error.

File: commands/run.js [hunk 5]
```diff
@@ -133,19 +133,19 @@ function cmd(bosco, args, allDone) {
...
 
     getRunList(function(err, runList) {
       if (err) return next(err);
-      var infraServices = _.filter(runList, function(i) { return i.service.type !== 'node'; });
+      var dockerServices = _.filter(runList, function(i) { return i.service.type !== 'node'; });
       var nodeServices = _.filter(runList, function(i) { return i.service.type === 'node' && _.startsWith('service-', i.name); });
       var nodeApps = _.filter(runList, function(i) { return i.service.type === 'node' && !_.startsWith('service-', i.name); });
       async.mapSeries([
-          {services: infraServices, type: 'general', limit: bosco.concurrency.cpu},
+          {services: dockerServices, type: 'docker', limit: bosco.concurrency.cpu},
```
Comment (geophree) [inline commands/run.js]: If we're going to say `type: 'docker'` we should limit it to `i.service.type === 'docker'`, not `i.service.type !== 'node'`.  Also, we may want to error if there are services that aren't docker or node.

File: commands/run.js (hunk 4 above)
Comment (bettiolo) [inline commands/run.js]: Is general services the same as docker services?

File: commands/run.js [hunk 6]
```diff
@@ -79,26 +79,41 @@ function cmd(bosco, args, allDone) {
...
   function startRunnableServices(next) {
+    var alreadyRunning = 0;
+
     function runService(runConfig, cb) {
       if (runConfig.service && runConfig.service.type === 'docker') {
         if (_.includes(runningServices, runConfig.name)) {
-          bosco.warn('Service ' + runConfig.name.green + ' is already running ...');
+          if (bosco.options.verbose) {
+            bosco.warn('Service ' + runConfig.name.green + ' is already running ...');
+          }
```
Comment (geophree) [inline commands/run.js]: Should we increment `alreadyRunning` here as well?

Comment (cliftonc) [inline commands/run.js]: Yes, will fix and push straight to master.

Comment (cliftonc) [inline commands/run.js]: Yes, will fix and push straight to master.

**Pull request merged**